apollo/lib/websocket/interest_type.py

```diff
@@ -0,0 +1,33 @@
+import enum
+
+from apollo.lib.decorators import with_db_session
+
+
+class WebSocketObserverInterestType(enum.Enum):
+    AGENT_LISTING = "agent_listing"
+
+    def run_corresponding_function(self):
+        return InterestTypeFunctionHandler().run_corresponding_function(
+            self
+        )
+
+
+class InterestTypeFunctionHandler:
+    def __init__(self):
+        self.type_function_mapping = {
```

Comment:
Doesn't need to be in the init I think. You should be able to just declare this as a property.

Reply:
decleration is possible, but get `TypeError: staticmethod object is not callable` once the method is called.